### PR TITLE
Fixes multiple devices on same SPI bus

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Fixes multiple devices on same SPI bus

## Motivation and Context

- Fixes nanoFramework/Home#1203
- Rework ESP32 to use manual chip select and allowing having multiple SPI devices on same bus
- Adjusted other hardware to support this scenario as well

## How Has This Been Tested?

On a real ESP32 device checking that Chip Select are working properly

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	// - Dockerfile.ESP32 = for ESP32 targets
 	// - Dockerfile.TI = for TI targets
 	// If you prefer, you can use the source files and adjust them they are located, with the same names in ./sources. This will alow you to customize them and add anything you may need on top.
-	"dockerFile": "Dockerfile.ALL",
+	"dockerFile": "Dockerfile.ESP32",
```
Comment (josesimoes) [inline .devcontainer/devcontainer.json:10]: Please revert this

File: targets/ESP32/_nanoCLR/System.Device.Spi/cpu_spi.cpp [hunk 2]
```diff
@@ -288,32 +323,55 @@ HRESULT CPU_SPI_Add_Device(const SPI_DEVICE_CONFIGURATION &spiDeviceConfig, uint
...
+    if (spiDeviceConfig.ChipSelectActive == 0)
+    {
+        initialPinValue = 1;
+    }
+
+    // Sets the CS pin as output
+    if (spiDeviceConfig.DeviceChipSelect >= 0)
+    {
+        // pinmode output
+        CPU_GPIO_EnableOutputPin(spiDeviceConfig.DeviceChipSelect, (GpioPinValue)initialPinValue, PinMode_Output);
```
Comment (josesimoes) [inline targets/ESP32/_nanoCLR/System.Device.Spi/cpu_spi.cpp:337]: Considering that above you're just toggling the CS pin, shouldn't you take care of setting it high or low here (according the config active low/high)?

Comment (Ellerbach) [reply targets/ESP32/_nanoCLR/System.Device.Spi/cpu_spi.cpp:337]: that's the "initialPinValue" doing this ;-)

Comment (josesimoes) [reply targets/ESP32/_nanoCLR/System.Device.Spi/cpu_spi.cpp:337]: OH!! I've missed that one. Maybe improve the naming of `ChipSelectActive` .... it's not immediately clear the intent. Something like `ChipSelectActiveLevel` ? or similar ...

Comment (Ellerbach) [reply targets/ESP32/_nanoCLR/System.Device.Spi/cpu_spi.cpp:337]: I can look, I wanted to keep the same name as in the rest of the HAL but while there, I can rename all of those.

**Review by josesimoes — APPROVED** [review]
LGTM! This will take care of it.
Thanks for fixing this and adding some improvements on the CPU SPI layer along the way! 💯

**Pull request merged**